src/test/java/com/google/api/generator/engine/writer/JavaWriterVisitorTest.java

```diff
@@ -201,6 +202,26 @@ public void writeVariableExpr_scopedStaticFinalDecl() {
     assertThat(writerVisitor.write()).isEqualTo("public static final boolean x");
   }
 
+  @Test
+  public void writeTernaryExpr_basic() {
+    Variable variable = Variable.builder().setName("x").setType(TypeNode.INT).build();
+    VariableExpr variableExpr = VariableExpr.builder().setVariable(variable).build();
+
+    Variable conditionVariable = Variable.builder().setName("condition").setType(TypeNode.BOOLEAN).build();
+    VariableExpr conditionExpr = VariableExpr.builder().setVariable(conditionVariable).build();
+    
+    Value value1 = PrimitiveValue.builder().setType(TypeNode.INT).setValue("3").build();
+    Expr thenExpr = ValueExpr.builder().setValue(value1).build();
+    Value value2 = PrimitiveValue.builder().setType(TypeNode.INT).setValue("4").build();
+    Expr elseExpr = ValueExpr.builder().setValue(value2).build();
+
+    TernaryExpr ternaryExpr = TernaryExpr.builder().setConditionExpr(conditionExpr).setThenExpr(thenExpr).setElseExpr(elseExpr).build();
+    AssignmentExpr assignExpr =
```

Comment:
Nit: Since this is a basic test, let's remove the assignment expression entirely.

Reply:
done.